packages/notifier/test/vat-integration/vat-pubsub.js

```diff
@@ -0,0 +1,27 @@
+import { Far } from '@endo/marshal';
+import { provide } from '@agoric/vat-data';
+import { vivifyDurablePublishKit } from '../../src/index.js';
+
+export const buildRootObject = (_vatPowers, vatParameters, baggage) => {
+  const makeDurablePublishKit = vivifyDurablePublishKit(
+    baggage,
+    'DurablePublishKit',
+  );
+  // TODO: Figure out why this breaks during upgrade!
+  const { publisher, subscriber } = provide(
+    baggage,
+    'publishKitSingleton',
+    () => makeDurablePublishKit(),
+  );
```

Comment:
I'm so sorry to take off with this incomplete, but I don't have enough gas left in the tank to figure out why it doesn't work as expected. The error is coming from [`makeRepresentative`](https://github.com/Agoric/agoric-sdk/blob/94625d38c3bb5333b00a69dd3086b1ac13490f62/packages/SwingSet/src/liveslots/virtualObjectManager.js#L694) during `reanimate` because `innerSelf` is a string rather than an object, and it occurs whether I use `provide` or manually invoke `baggage.get` after a conditional `baggage.init`. I'm almost certain that I'm just trying to do something a little incorrectly.

<details><summary><code>yarn test test/test-publish-kit.js</code></summary>

```
error during vat dispatch() of startVat,[object Object] (TypeError#1)
TypeError#1: Object.getOwnPropertyNames: invalid object
TypeError: Object.getOwnPropertyNames: invalid object
 at getOwnPropertyNames ()
 at makeRepresentative (/bundled-source/.../packages/SwingSet/src/liveslots/virtualObjectManager.js:694)
 at reanimate (/bundled-source/.../packages/SwingSet/src/liveslots/virtualObjectManager.js:770)
 at convertSlotToVal (/bundled-source/.../packages/SwingSet/src/liveslots/liveslots.js:771)
 at decodeSlotCommon (/bundled-source/.../node_modules/@endo/marshal/src/marshal.js:254)
 at decodeFromCapData (/bundled-source/.../node_modules/@endo/marshal/src/encodeToCapData.js:414)
 at decodeEntry (/bundled-source/.../node_modules/@endo/marshal/src/encodeToCapData.js:483)
 at map ()
 at decodeFromCapData (/bundled-source/.../node_modules/@endo/marshal/src/encodeToCapData.js:485)
 at unserialize (/bundled-source/.../node_modules/@endo/marshal/src/marshal.js:359)
 at runWithoutMetering (/bundled-source/.../packages/SwingSet/src/supervisors/subprocess-xsnap/supervisor-subprocess-xsnap.js:52)
 at buildRootObject (.../notifier/test/vat-integration/vat-pubsub.js:14)
 at startVat (/bundled-source/.../packages/SwingSet/src/liveslots/liveslots.js:1418)
 at ()
 at ()

UnhandledPromiseRejectionWarning: (TypeError#1)
Logging sent error stack (RemoteError#1)
RemoteError#1: vat-upgrade failure
Error: vat-upgrade failure
 at construct ()
 at Error (/bundled-source/.../node_modules/ses/src/error/tame-error-constructor.js:56)
 at makeError (/bundled-source/.../node_modules/ses/src/error/assert.js:242)
 at decodeErrorCommon (/bundled-source/.../node_modules/@endo/marshal/src/marshal.js:284)
 at decodeFromCapData (/bundled-source/.../node_modules/@endo/marshal/src/encodeToCapData.js:425)
 at map ()
 at map ()
 at unserialize (/bundled-source/.../node_modules/@endo/marshal/src/marshal.js:359)
 at deliver (/bundled-source/.../packages/SwingSet/src/liveslots/liveslots.js:1053)
 at dispatchToUserspace (/bundled-source/.../packages/SwingSet/src/liveslots/liveslots.js:1452)
 at runWithoutMetering (/bundled-source/.../packages/SwingSet/src/supervisors/subprocess-xsnap/supervisor-subprocess-xsnap.js:52)
 at ()

RemoteError#1 ERROR_NOTE: Sent as error:liveSlots:v2#70001
Logging sent error stack (RemoteError(error:liveSlots:v2#70001)#1)
RemoteError(error:liveSlots:v2#70001)#1: vat-upgrade failure
Error: vat-upgrade failure
 at construct ()
 at Error (/bundled-source/.../node_modules/ses/src/error/tame-error-constructor.js:56)
 at makeError (/bundled-source/.../node_modules/ses/src/error/assert.js:242)
 at decodeErrorCommon (/bundled-source/.../node_modules/@endo/marshal/src/marshal.js:284)
 at decodeFromCapData (/bundled-source/.../node_modules/@endo/marshal/src/encodeToCapData.js:425)
 at unserialize (/bundled-source/.../node_modules/@endo/marshal/src/marshal.js:359)
 at notifyOnePromise (/bundled-source/.../packages/SwingSet/src/liveslots/liveslots.js:1189)
 at notify (/bundled-source/.../packages/SwingSet/src/liveslots/liveslots.js:1213)
 at dispatchToUserspace (/bundled-source/.../packages/SwingSet/src/liveslots/liveslots.js:1457)
 at runWithoutMetering (/bundled-source/.../packages/SwingSet/src/supervisors/subprocess-xsnap/supervisor-subprocess-xsnap.js:52)
 at ()

RemoteError(error:liveSlots:v2#70001)#1 ERROR_NOTE: Sent as error:liveSlots:v1#70001
REJECTED from ava test: {
  body: '{"@qclass":"error","errorId":"error:liveSlots:v1#70001","message":"vat-upgrade failure","name":"Error"}',
  slots: []
}
  ✖ durable publish kit upgrade trauma (full-vat integration) Rejected promise returned by test
  ─

  durable publish kit upgrade trauma (full-vat integration)

  Rejected promise returned by test. Reason:

  {
    body: '{"@qclass":"error","errorId":"error:liveSlots:v1#70001","message":"vat-upgrade failure","name":"Error"}',
    slots: [],
  }
```

</details>

Reply:
Does this work with the LRU fix in place? Or is the TODO now stale?